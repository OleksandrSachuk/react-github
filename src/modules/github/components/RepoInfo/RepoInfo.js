import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

const propTypes = {
  repoInfo: PropTypes.object.isRequired,
};

const RepoInfo = ({ repoInfo }) => (
  <Fragment>
    { repoInfo.id
      ? <div>
        <div>Github user selected repo information:</div>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Name</TableCell>
              <TableCell>Description</TableCell>
              <TableCell>Fork</TableCell>
              <TableCell>Stars</TableCell>
              <TableCell>Issues</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
            <TableCell>{ repoInfo.name }</TableCell>
            <TableCell>{ repoInfo.description }</TableCell>
            <TableCell>{ repoInfo.fork.toString() }</TableCell>
            <TableCell>{ repoInfo.stargazers_count }</TableCell>
            <TableCell>{ repoInfo.open_issues }</TableCell>
          </TableRow>
        </TableBody>
      </Table>
      </div>
      : null
    }
  </Fragment>
);

RepoInfo.propTypes = propTypes;

export default RepoInfo;
