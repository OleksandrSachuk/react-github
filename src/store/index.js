import { createStore, applyMiddleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import rootEpic from '../modules/github/epics';

import rootReducer from '../modules/github/reducers';

const epicMiddleware = createEpicMiddleware();

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  applyMiddleware(epicMiddleware),
);

epicMiddleware.run(rootEpic);

export default store;