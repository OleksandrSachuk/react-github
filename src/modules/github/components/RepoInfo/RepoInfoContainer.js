import { connect } from 'react-redux';

import RepoInfo from './RepoInfo';

const mapStateToProps = (state) => ({
  repoInfo: state.user.repoInfo,
});

export default connect(
  mapStateToProps,
)(RepoInfo);

