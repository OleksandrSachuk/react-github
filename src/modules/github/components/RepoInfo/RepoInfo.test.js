import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RepoInfo from './';
import store from '../../../../store';

it('renders RepoInfo without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RepoInfo store={ store }/>, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('RepoInfo snapshot test', () => {

  const component = renderer.create(
    <RepoInfo store={ store }/>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
