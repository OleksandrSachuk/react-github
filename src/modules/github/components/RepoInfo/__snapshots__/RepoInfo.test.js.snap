// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoInfo snapshot test 1`] = `null`;
